### ISSUE TITLE
Support torch::jit::script::Module converting to torch::nn::Module.

### ISSUE THREAD

**Issue by crissallan** [pytorch/pytorch#51328]
## 🚀 Feature
Support torch::jit::script::Module converting to torch::nn::Module.

## Motivation
In some cases like image-style transfer, we need to calculate the gradient of loss w.r.t the input tensor. Howerver, if I load a pre-trained Pytorch CNN in C++ by using jit::script::Module. The pre-trained CNN of the jit::script::Module can't pass the gradients to the input tensor. Because it seems the jit::script::Module doesn't support the AutoGrad.

So if there is a convenient way to convert the pre-trained jit::script::Module into nn::Module. Then we could use AutoGrad of the pre-trained model more conveniently!

## Pitch

None

## Alternatives

None

## Additional context

None

**Comment by ssbotelh** [comment]
Hi there, was wondering if there was any progress on this effort.
If not, does anybody know if there are plans to support `jit::script::Module` to `nn::Module` conversion in the near future? Thanks!

**Comment by KongCang** [comment]
I have the same problem as well

**Comment by BlessingDev** [comment]
I also stuck in similar problem. I trained my model in c++ and tried to inference it in pytorch. However pytorch failed to load the model serialized in c++. I expect this problem can be aslo solved with this issue. Probably by converting `nn::Module` to `jit::script::Module` and save script model.

This issue is already pended two years ago and still not solved. I seems not to be able to expect this issue to be resolved in the near future. But really looking forward to be solved.

**Comment by Alone-wl** [comment]
> expect this issue to be resolved in the near future. But really looking forward to be solved.

Has pended four years.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
